### PR TITLE
deps: Bump `envoy.code.check` -> 0.2.1

### PR DESCRIPTION
**Pull request by phlax**
and update `extensions_schema.yaml`

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
Currently the descriptions for the extensions metadata are hardcoded in `protodoc` and also added as comments in the extensions schema config

This extends the extensions schema config to include the descriptions. The hardcoded values will be removed from protodoc subsequently and it can use the configured values.

File: tools/extensions/extensions_schema.yaml [hunk 1]
```diff
@@ -94,11 +103,13 @@ categories:
...
-# This extension is stable and is expected to be production usable.
-- stable
-# This extension is functional but has not had substantial production burn
-# time, use only with this caveat.
-- alpha
-# This extension is work-in-progress. Functionality is incomplete and it is
-# not intended for production use.
-- wip
+- name: stable
+  # This extension is stable and is expected to be production usable.
```
Comment (phlax) [inline tools/extensions/extensions_schema.yaml:107]: description is ommitted here as that is the case in `protodoc`

**Review by htuch — APPROVED** [review]
Thanks - prefer in future PRs that changes which don't have an obvious connection are split into separate PRs, but I guess this is a "improve protodoc" arc?

**Comment by phlax** [comment]
> prefer in future PRs that changes which don't have an obvious connection

there are 2 changes here:

- data schema
- parser of ^^

the changes were inseparable and related to nothing other than each other

**Pull request merged**